### DISCUSSION TITLE
No Data in HA

### DISCUSSION THREAD

**Discussion by roethigj** [s-allius/tsun-gen3-proxy#55] [Q&A] [answered]
Hi,

thank you for the proxy. 

Unfortunately I can't get any data in HA.

I use a MS600 GEN3 Inverter (R17....). The container is up and running. The DNS is resolved to the proxy. I receive values for the "Controller". But I don't get any data for "Micro Inverter" or "Module PV 1".

This is what I get from the logs:
```
'2024-04-19 10:51:32  INFO | root | Server "tsun-proxy - " will be started'
'2024-04-19 10:51:32  INFO | root | Initialize proxy statistics'
'2024-04-19 10:51:32  INFO | root | Scheduler init'
/usr/local/lib/python3.12/site-packages/tzlocal/unix.py:193: UserWarning: Can not find any timezone configuration, defaulting to UTC.
  warnings.warn("Can not find any timezone configuration, defaulting to UTC.")
'2024-04-19 10:51:32  INFO | mqtt | start MQTT: host:192.168.178.101  port:1883  user:MQTT_User'
'2024-04-19 10:51:32  INFO | mqtt | MQTT broker connection established'
'2024-04-19 10:51:32  INFO | root | Initialize proxy device on home assistant'
'2024-04-19 10:52:01  INFO | root | Accept connection from  ('192.168.178.55', 55007)'
'2024-04-19 10:52:06  INFO | root | Connected to ('logger.talent-monitoring.com', 5005)'
'2024-04-19 10:54:15  INFO | root | [Errno 110] Connect call failed ('47.254.132.252', 5005)'
'2024-04-19 10:54:15  INFO | root | Connected to ('logger.talent-monitoring.com', 5005)'
```
[trace.log](https://github.com/s-allius/tsun-gen3-proxy/files/15039374/trace.log)

Any idea what I am missing?

Thanks and regards,
Jo

**Comment by s-allius** [comment]
Hi Jo,
let's see. I don't see anything conspicuous in the trace for now. The fact that the timezone is not set correctly should not really matter, as we are not yet sending timestamps to HA.
Have you looked at the trace files of the Docker container? There should be two files in the log directory. There may be a hint there...

What will happen when you restart HA? You should see a message for that in the proxy log.
Regards Stefan

**Comment by s-allius** [comment, marked as answer]
Hi Jo,
I oversaw the log.file. In the log I never see a reply from the TSUN Cloud. That might be a big problem..

You can disable the connection to the TSUN cloud in your config.toml file, by setting TSUN.enabled to false. So we never send pakets to the internet and don't wait for the reply. 

In detail: In the trace you can see:
'2024-04-19 09:33:17  INFO |  msg | >   Ctl: 0x91 Msg: 'msg_get_time''
'2024-04-19 09:33:17  INFO |  msg |  >> Ctl: 0x91 Msg: 'msg_get_time''

that the inverter request the time, but never get an answer. That might the route cause. You can also see, taht the proxy sends an request for the time to the TSUN cloud, and also get no reply.

Pls check if your docker container has internet access

**Comment by roethigj** [comment]
Hello Stefan,

yes, this was the problem. The container had no connection to logger.talent-monitoring.com.
The reason was a wrong configuration of my router.

Maybe, as an enhancement, you can set up the proxy to work without answers from talent-monitoring, as you do with "tsun.eneabled = false", but give a warning.
I remember the days when talent-monitoring server was not available. It would be a pity if this breaks the proxy.

Again: Thank you for the work.

Regards
Jo

**Reply by s-allius** [reply]
Hi Jo,

thats a good point. Normally the proxy will work without a connection to TSUN. Only for the first two messages there is a the answer needed. I will have a look, if I can improve it.
